File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/ErdOnlineQueryService.java
package org.javamaster.invocationlab.admin.service;

import org.javamaster.invocationlab.admin.model.erd.AesReqVo;
import org.javamaster.invocationlab.admin.model.erd.Column;
import org.javamaster.invocationlab.admin.model.erd.CommonErdVo;
import org.javamaster.invocationlab.admin.model.erd.QueryReqVo;
import org.javamaster.invocationlab.admin.model.erd.SaveQueryReqVo;
import org.javamaster.invocationlab.admin.model.erd.SqlExecResVo;
import org.javamaster.invocationlab.admin.model.erd.Table;
import org.javamaster.invocationlab.admin.model.erd.TokenVo;
import org.javamaster.invocationlab.admin.model.erd.Tree;
import com.alibaba.fastjson.JSONArray;
import com.alibaba.fastjson.JSONObject;
import org.apache.commons.lang3.tuple.Triple;
import org.springframework.http.MediaType;

import java.util.List;

/**
 * @author yudong
 */
public interface ErdOnlineQueryService {
    JSONArray load(String projectId) throws Exception;

    List<Tree> getQueryTreeList(String projectId) throws Exception;

    Boolean createDirOrQuery(QueryReqVo reqVo, TokenVo tokenVo) throws Exception;

    Tree queryTree(String treeNodeId) throws Exception;

    Integer deleteQueryTree(String projectId, String treeNodeId) throws Exception;

    JSONObject queryHistory(CommonErdVo reqVo, TokenVo tokenVo);

    Tree saveQueryTree(SaveQueryReqVo reqVo, TokenVo tokenVo) throws Exception;

    SqlExecResVo execSql(CommonErdVo jsonObjectReq, TokenVo tokenVo) throws Exception;

    List<SqlExecResVo> execUpdate(CommonErdVo jsonObjectReq, TokenVo tokenVo) throws Exception;

    List<String> getDbs(String projectId, TokenVo tokenVo) throws Exception;

    List<Table> getTables(CommonErdVo reqVo, TokenVo tokenVo) throws Exception;

    List<Column> getTableColumns(CommonErdVo reqVo, TokenVo tokenVo) throws Exception;

    Triple<String, MediaType, byte[]> exportSql(CommonErdVo jsonObjectReq, TokenVo tokenVo) throws Exception;

    String aes(AesReqVo reqVo, TokenVo tokenVo) throws Exception;

    Integer getTableRecordTotal(CommonErdVo reqVo, TokenVo tokenVo) throws Exception;
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/ErdOnlineQueryService.kt
package org.javamaster.invocationlab.admin.service

import org.javamaster.invocationlab.admin.model.erd.*
import com.alibaba.fastjson.JSONArray
import com.alibaba.fastjson.JSONObject
import org.apache.commons.lang3.tuple.Triple
import org.springframework.http.MediaType

/**
 * @author yudong
 */
interface ErdOnlineQueryService {

    fun load(projectId: String): JSONArray


    fun getQueryTreeList(projectId: String): MutableList<Tree>


    fun createDirOrQuery(reqVo: QueryReqVo, tokenVo: TokenVo): Boolean


    fun queryTree(treeNodeId: String): Tree


    fun deleteQueryTree(projectId: String, treeNodeId: String): Int

    fun queryHistory(reqVo: CommonErdVo, tokenVo: TokenVo): JSONObject


    fun saveQueryTree(reqVo: SaveQueryReqVo, tokenVo: TokenVo): Tree


    fun execSql(jsonObjectReq: CommonErdVo, tokenVo: TokenVo): SqlExecResVo


    fun execUpdate(jsonObjectReq: CommonErdVo, tokenVo: TokenVo): List<SqlExecResVo>


    fun getDbs(projectId: String, tokenVo: TokenVo): List<String>


    fun getTables(reqVo: CommonErdVo, tokenVo: TokenVo): List<Table>


    fun getTableColumns(reqVo: CommonErdVo, tokenVo: TokenVo): List<Column>


    fun exportSql(jsonObjectReq: CommonErdVo, tokenVo: TokenVo): Triple<String, MediaType, ByteArray>


    fun aes(reqVo: AesReqVo, tokenVo: TokenVo): String


    fun getTableRecordTotal(reqVo: CommonErdVo, tokenVo: TokenVo): Int
}
